submariner-operator/submariner-metrics-redesign.md

```diff
@@ -16,43 +16,46 @@ This enhancement proposes a redesign of existing metrics for Submariner Gateway
 service. Pods running on the same node can access ports without requiring pods to be opened in the cloud infra. Any queries
 for these metrics will land on the proxy which will always run on the same node.
 
-Note that this proposal doesn't address the issue where metrics port is unavailable on the node. That should be addressed as
-a separate enhancement.
+This proposal also recommends changing the port used by metrics http servers in gateway and globalnet pods to a free range
+port to avoid problem of port being already in use on host network.
 
 ## Design Details
 
-* A new `status.submariner.io/gateway=active` label will be added to gateway node that is currently active.
-* Submarine RouteAgent will set the label on the relevant Node.
-* A new DaemonSet `submariner-metrics` will be created. It will use the NodeSelector `status.submariner.io/gateway=active`
-to make sure it always runs on the active gateway node.
+* A new DaemonSet `submariner-metrics-proxy` will be created. It will use the NodeSelector `submariner.io/gateway=true`
+to run on all gateway nodes, active or not.
 * This DaemonSet will use `schedulerName: default-scheduler` to make sure that metrics proxy pod moves with the active gateway
 label.
 * DaemonSet will also use toleration to make sure it can always be scheduled on gateway node, irrespective of any taints.
-* `submariner-metrics` Pod will use `netcat` to act as proxy and forward all incoming requests on metrics port to the
+* `submariner-metrics-proxy` Pod will use `netcat` to act as proxy and forward all incoming requests on metrics port to the
 metrics port on the node IP.
 * The proxy pod will run two containers if globlanet is enabled, one each for gateway and globalnet.
 * Existing metrics service for gateway and globalnet will be modified to use the new metrics proxy pod. This will be
-done by using same label selector as metrics DaemonSet.
+done by using same label selector as metrics proxy DaemonSet. This will make sure users can continue using existing services
+and ports without needing changes.
+* `submariner-gateway` will be modified to use port `32780` instead of `8080` and `submariner-globalnet` will be
+modified to use `32781` instead of `8081`. A new env variable will be used for these ports.
+* [Optional] Modify `subctl cloud prepare` to no longer open metrics ports.
```

Comment:
:+1: 
There will be no longer need for this, and keeping dead code will serve no purpose.